### PR TITLE
GSP-139: Split Specs

### LINKED ISSUE #138: Move specs back into go-storage
rs-storage maybe need to maintain their own definitions and don't need to have the same set with golang. 

So how about move specs back into go-storage? After this change, we don't need to update `go.mod` anymore.

### PR DESCRIPTION
**Pull request by Xuanwo**
Signed-off-by: Xuanwo <github@xuanwo.io>

### CODE REVIEW

**Review by JinnyYi — APPROVED** [review]

File: rfcs/139-split-specs.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+> We cannot distinguish 2 LGTMs for RFCs and 1 LGTM for other PRs automatically?
+
+For now, the restriction is executed by committer / maintiner mannully. In the further, we will assign labels like `need-2-lgtm`.
+
+## Compatibility
+
+To not break our links to existing proposals, we will:
+
+- Move all issues into `go-storage`
```
Comment (xxchan) [inline rfcs/139-split-specs.md:74]: How will the PR numbers and PR links be affected?

File: rfcs/139-split-specs.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+So I propose to split specs into projects:
+
+- Move `/definitions` and `/go` into `go-storage`.
+- Move `/rfcs` and `/spec` into `go-storage`.
+- Move `/rust` into `rs-storage`.
+
+After those changes, we can be more focused on building up features instead of caring about behavior that across languages. And project's committer and maintainer will have the ability to approved proposals without contact other teams.
+
+## Rationale
```
Comment (xxchan) [inline rfcs/139-split-specs.md:35]: Why to split `/definitions` is not stated here

File: rfcs/139-split-specs.md (hunk 1 above)
Comment (Xuanwo) [reply rfcs/139-split-specs.md:74]: They will be assigned new numbers just like create new issues in `go-storage`.

**Comment by Xuanwo** [comment]
ping @xxchan to take a look.

**Review by xxchan — APPROVED** [review]

**Pull request merged**